lib/config_files/grub2/default.rb

```diff
@@ -0,0 +1,176 @@
+require "config_files/base_model"
+require "config_files/augeas_parser"
+
+module ConfigFiles
+  module Grub2
+    # Represents grub configuration in /etc/default/grub
+    # Main features:
+    #
+    # - Do not overwrite files
+    # - When setting value first try to just change value if key already exists
+    # - When key is not set, then try to find commented out line with key and
+    #   replace it with real config
+    # - When even commented out code is not there, then append configuration
+    #   to the end of file
+    class Default < BaseModel
+      PARSER = AugeasParser.new("sysconfig.lns")
+      PATH = "/etc/default/grub"
+
+      def initialize(file_class: File)
+        super(PARSER, PATH, file_class: file_class)
+        self.data = AugeasTree.new
+      end
+
+      def os_prober
+        @os_prober ||= BooleanValue.new(
+          "GRUB_DISABLE_OS_PROBER",
+          self,
+          default:     true,
+          # grub key is disable, so use reverse logic
+          true_value:  "false",
+          false_value: "true"
+        )
+      end
+
+      def disable_recovery_entry
+        generic_set("GRUB_DISABLE_RECOVERY", "true")
+      end
+
+      def enable_recovery_entry(kernel_params)
+        generic_set("GRUB_DISABLE_RECOVERY", "false")
+        generic_set("GRUB_CMDLINE_LINUX_RECOVERY", kernel_params)
+      end
+
+      def timeout
+        data["GRUB_TIMEOUT"] || 10
+      end
+
+      def timeout=(value)
+        generic_set("GRUB_TIMEOUT", value)
+      end
+
+      def cryptodisk
+        @cryptodisk ||= BooleanValue.new(
+          "GRUB_ENABLE_CRYPTODISK",
+          self
+        )
+      end
+
+      def terminal
+        case data["GRUB_TERMINAL"]
+        when "console", "", nil
+          :console
+        when "serial"
+          :serial
+        when "gfxterm"
+          :gfxterm
+        else
+          raise "unknown GRUB_TERMINAL option #{data["GRUB_TERMINAL"].inspect}"
+        end
+      end
+
+      VALID_TERMINAL_OPTIONS = [:serial, :console, :gfxterm]
+      def terminal=(value)
+        if !VALID_TERMINAL_OPTIONS.include?(value)
+          raise ArgumentError, "invalid value #{value.inspect}"
+        end
+
+        generic_set("GRUB_TERMINAL", value.to_s)
+      end
+
+      def serial_console=(value)
+        self.terminal = :serial
+        generic_set("GRUB_SERIAL_COMMAND", value)
+      end
+
+      def serial_console
+        data["GRUB_SERIAL_COMMAND"]
+      end
+
+      # powerfull low level method that sets any value in grub config.
+      # @note prefer to use specialized methods
+      def generic_set(key, value)
```

Comment:
Sounds like something to extract to a more general place. "Modify or uncomment or add" sounds like a general pattern to me, not something grub-specific.